doozerlib/rpmcfg.py

```diff
@@ -85,32 +84,73 @@ def push_tag(self):
             raise ValueError('Must run set_nvr() before calling!')
 
         with Dir(self.source_path):
-            exectools.cmd_assert('git push origin {}'.format(self.tag), retries=3)
+            exectools.cmd_assert('git push origin --tags', retries=3)
 
-    def commit_changes(self):
+    def commit_changes(self, scratch):
         with Dir(self.source_path):
+            if self.config.content.build.use_source_tito_config:
+                # just use the tito tagger to change spec and tag
+                exectools.cmd_assert(["tito", "tag", "--no-auto-changelog",
+                    "--use-version", self.version,
+                    "--use-release", "{}%{{?dist}}".format(self.release)
+                ])
+                if self.config.content.build.push_release_commit and not scratch:
+                    exectools.cmd_assert("git push origin")
+                return
+
             exectools.cmd_assert("git add .")
-            exectools.cmd_assert(['git', 'commit', '-m', "Local commit for dist-git build"])
+            commit_msg = "Automatic commit of package [{name}] release [{version}-{release}].".format(
+                name=self.config.name, version=self.version, release=self.release
+            )
+            exectools.cmd_assert(['git', 'commit', '-m', commit_msg])
 
-    def rollback_changes(self):
+    def post_build(self, scratch):
+        build_spec = self.config.content.build
         with Dir(self.source_path):
-            exectools.cmd_assert("git reset --hard HEAD~")
-            exectools.cmd_assert("git tag -d {} {}".format(self.tag, self.vtag))
+            if self.build_status and not scratch:
+                # success; push the tag
+                try:
+                    self.push_tag()
+                except Exception:
+                    raise RuntimeError('Build succeeded but failure pushing RPM tag for {}'.format(self.qualified_name))
+
+            elif build_spec.use_source_tito_config:
+                # don't leave the tag/commit lying around if not a valid build
+                exectools.cmd_assert("tito tag --undo")
+
+            if not build_spec.push_release_commit and not build_spec.use_source_tito_config:
+                # temporary tag/commit; never leave lying around
+                exectools.cmd_assert("git reset --hard HEAD~")
+                exectools.cmd_assert("git tag -d {}".format(self.tag))
 
     def tito_setup(self):
+        if self.config.content.build.use_source_tito_config:
+            return  # rely on tito already set up in source
+
         tito_dir = os.path.join(self.source_path, '.tito')
+        tito_target = self.config.content.build.tito_target
+        tito_target = tito_target if tito_target else 'aos'
+        tito_dist = self.config.content.build.tito_dist
+        tito_dist = tito_dist if tito_dist else '.el7aos'
```

Comment:
hard coded rhel7, should be pulled from group.yaml

Reply:
just for the default :)

I agree this ought to come from group.yaml -- all of these new attrs probably should. Just wasn't ready to introduce that extra bit of logic for the immediate fix. I guess these could be properties of `RPMMetadata`...